### PR TITLE
Issue 18, part1: profiles per channel

### PR DESCRIPTION
**Pull request by Hejwo**

### CODE REVIEW

File: generators/setup-docker/configTransformers.js [hunk 1]
```diff
@@ -156,8 +160,11 @@ function transformChannelConfig(channelJsonFormat, orgsTransformed) {
 
   return {
     key: channelJsonFormat.key,
-    name: channelJsonFormat.name,
+    name: channelName,
     orgs: orgsForChannel,
+    profile: {
+      name: _.chain(channelName).camelCase().upperFirst().value(),
```
Comment (dzikowski) [inline generators/setup-docker/configTransformers.js:166]: I would create a separate variable for this, for example `const profileName` right after `const channelName`

File: generators/setup-docker/templates/fabric-config/configtx.yaml [hunk 2]
```diff
@@ -44,25 +47,28 @@ Organizations:
...
       Name: <%= org.mspName %>
       ID: <%= org.mspName %>
       MSPDir: crypto-config/peerOrganizations/<%= org.domain %>/msp
-<% if(isHlf20) { %>
+
       Policies:
           Readers:
               Type: Signature
-              Rule: "OR('<%= org.mspName %>.admin', '<%= org.mspName %>.peer', '<%= org.mspName %>.client')"
+              Rule: "OR('<%= org.mspName %>.member')"
```
Comment (dzikowski) [inline generators/setup-docker/templates/fabric-config/configtx.yaml:61]: 👍

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 3]
```diff
@@ -126,11 +126,11 @@ function stopNetwork() {
 
 function generateChannelsArtifacts() {
   <% if (!channels || !channels.length) { -%>
-    echo "No channels"
+      echo "No channels"
```
Comment (dzikowski) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:129]: why? ;-)

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 4]
```diff
@@ -126,11 +126,11 @@ function stopNetwork() {
...
   <% } else { -%>
     <% channels.forEach(function(channel){  -%>
-      printHeadline "Generating config for '<%= channel.name %>'" "U1F913"
-      createChannelTx "<%= channel.name %>" "$FABRICA_NETWORK_ROOT/fabric-config" "AllOrgChannel" "$FABRICA_NETWORK_ROOT/fabric-config/config"
+    printHeadline "Generating config for '<%= channel.name %>'" "U1F913"
+    createChannelTx "<%= channel.name %>" "$FABRICA_NETWORK_ROOT/fabric-config" "<%= channel.profile.name %>" "$FABRICA_NETWORK_ROOT/fabric-config/config"
```
Comment (dzikowski) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:133]: why the whitespaces from the beginning are removed?

File: generators/setup-docker/templates/fabric-config/configtx.yaml [hunk 5]
```diff
@@ -126,32 +130,52 @@ Orderer: &OrdererDefaults
         # from the orderer for the peer to validate it.
         BlockValidation:
             Type: ImplicitMeta
-            Rule: "ANY Writers"<% } %>
+            Rule: "ANY Writers"
+<% if(!isHlf20) { %>
+    Capabilities:
+        <<: *OrdererCapabilities
+<% } %>
```
Comment (dzikowski) [inline generators/setup-docker/templates/fabric-config/configtx.yaml:137]: you can use `-%>` to avoid additional blank lines

Comment (dzikowski) [reply generators/setup-docker/templates/fabric-config/configtx.yaml:137]: (see the rest of this file as well)

File: generators/setup-docker/configTransformers.js (hunk 1 above)
Comment (dzikowski) [inline generators/setup-docker/configTransformers.js:166]: I would create a separate variable for that, for example `const profileName` just after `const channelName`

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh (hunk 3 above)
Comment (dzikowski) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:129]: why the additional whitespace?

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 6]
```diff
@@ -126,11 +126,11 @@ function stopNetwork() {
 
 function generateChannelsArtifacts() {
   <% if (!channels || !channels.length) { -%>
-    echo "No channels"
+      echo "No channels"
   <% } else { -%>
     <% channels.forEach(function(channel){  -%>
-      printHeadline "Generating config for '<%= channel.name %>'" "U1F913"
-      createChannelTx "<%= channel.name %>" "$FABRICA_NETWORK_ROOT/fabric-config" "AllOrgChannel" "$FABRICA_NETWORK_ROOT/fabric-config/config"
```
Comment (dzikowski) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:133]: why the whitespaces are removed? ;-)

**Review by dzikowski — APPROVED** [review]
lgtm

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh (hunk 3 above)
Comment (Hejwo) [reply generators/setup-docker/templates/fabric-docker/commands-generated.sh:129]: poor merge :)

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh (hunk 4 above)
Comment (Hejwo) [reply generators/setup-docker/templates/fabric-docker/commands-generated.sh:133]: poor merge :)

**Pull request merged**